### DISCUSSION TITLE
Target tags vs capabilities

### DISCUSSION THREAD

**Discussion by agluszak** [build-server-protocol/build-server-protocol#489] [General]
Target tags and capabilities are not fully orthogonal. Why do we have a test tag *and* a test capability? Should benchmarks be "tested", or should they be run? What does it mean to debug a library?

I think this issue comes from the fact that we haven't clearly defined what a "target" is. See #470

**Comment by jastice** [comment]
the `test` tag came from the need to map the sbt test _configuration_ back to IntelliJ test _scope_ - whereas `canTest` does not carry such an association and is simply any target that you can run tests on. In practice, of course this overlaps, at least in the case of sbt and related tools. We can have a discussion if the concept of capabilities can be made more general or streamlined.
